doc/ref/gappkg.xml

```diff
@@ -173,5 +184,2074 @@ users.
+conditions will be checked when the package will be loaded 
+(see <Ref Sect="Version Numbers"/>). 
+Since there are no requirements that have to be tested, 
+<C>AvailabilityTest</C> just uses <Ref Func="ReturnTrue"/>.
+<P/>
+Now start &GAP; (without using the <C>-r</C> option) and the <F>.gap</F>
+directory will be added to the &GAP; root directory to
+allow &GAP; to find the packages installed there
+(see <Ref Sect="GAP Root Directories"/>).
+<P/>
+<Log><![CDATA[
+gap> LoadPackage("test");
+true
+]]></Log>
+<P/>
+This &GAP; package is too simple  to be useful, but we have succeeded
+in loading it via <Ref Func="LoadPackage"/>, satisfying 
+all specified dependencies.
+<P/>
+You can also try to create a basic &GAP; package with the help of the
+tool called <Package>PackageMaker</Package>, created by Max Horn and
+available at <URL>https://github.com/gap-system/PackageMaker</URL>. The
+<Package>PackageMaker</Package> asks several questions about the intended
+package and then creates a new directory for it and populates it with all
+the files needed for a basic package.
+
+</Section>
+
+
+<!-- %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% -->
+<Section Label="File Structure">
+<Heading>File Structure</Heading>
+
+Package files may follow the style used for the &GAP; library.
+Every file in the &GAP; library starts with a header that lists the 
+filename, copyright, a short description of the file contents and the 
+original authors of this file, and ends with a comment line <C>#E</C>. 
+Indentation in functions and the use of decorative spaces in the code 
+are left to the decision of the authors of each file. Global (i.e. 
+re-used elsewhere) comments usually are indented by two hash marks and 
+two blanks, in particular, every declaration or method or function 
+installation which is not only of local scope is separated by a header.
+<P/>
+Historically, when the &GAP; main manuals were based on the &TeX; macros 
+described in the document <Q>The gapmacro.tex Manual Format</Q>
+(the file <F>gap4rN/doc/gapmacrodoc.pdf</F>)
+such headers were used for the manuals and have the type
+<P/>
+<Log><![CDATA[
+#############################################################################
+##
+#X  ExampleFunction(<A>,<B>)
+##
+##  This function does great things.
+]]></Log>
+<P/>
+where <Q>X</Q> was one of the letters: <C>F</C>, <C>A</C>, <C>P</C>, <C>O</C>,
+<C>C</C>, <C>R</C> or <C>V</C> indicating whether the object declared will
+be a function, attribute, property, operation, category, representation or 
+variable, respectively. Additionally <C>M</C> was used in <C>.gi</C> files 
+for method installations. Then a sample usage of the function was given,
+followed by a comment which described the identifier.
+This description was automatically be extracted to build the
+manual source, if there is a <C>\Declaration</C> line in some
+<C>.msk</C> file together with an appropriate configuration file.
+<P/>
```

Comment:
Personally, I'd remove the stuff between `Historically`and this point. Well, or perhaps instead move it to a new "historical" section (the information above is useful for understanding why some old code has comments like that). But other than that historical interest, I find it quite distracting. Prospective package authors have to slog through all of that for no good reason before they get to the stuff that actually matters for them.